src/github3/repos/repo.py

```diff
@@ -557,6 +558,52 @@ def contributors(self, anon=False, number=-1, etag=None):
             params = {"anon": "true"}
         return self._iter(int(number), url, users.Contributor, params, etag)
 
+    def views(self, per='day'):
+        """Get the total number of repository views and breakdown per day or
+        week for the last 14 days.
+
+        .. versionadded:: 1.4.0
+
+        See also: https://developer.github.com/v3/repos/traffic/
+
+        :param str per:
+            (optional), ('day', 'week'), views reporting period. Default 'day'
+            will return views per day for the last 14 days.
+        :returns:
+            views data
+        :rtype:
+            :class:`~github3.repos.traffic.ViewsStats`
+        """
+        params = {}
+        if per in ("day", "week"):
+            params.update(per=per)
```

Comment:
What happens if a user requests a `per` that is not one of day or week? There should probably be some checking around this and a raise if invalid options are passed.

Looking at the API itself, it does require a `per` argument, so it should be enough to validate that what the function was sent was one of `day` or `week`, and then just pass whatever it is into the params. 

Maybe here, just have an `else` block that raises an error about invalid arguments.

Reply:
Sounds reasonable. Will do.

Reply:
Although, to be consistent with this behavior, you'll want to update `create_blob()`, `create_status()`, `create_tag()` and `archive()` in the `Repository` class as well. All these functions accept parameters with a string value from some allowed set, and don't fail if the value is outside of that set (just silently ignore the parameter).

Reply:
That'd be a nice follow up PR.

Or maybe we can just follow the example. Presumably the API itself will return the error back to the user, and github3.py will surface it accordingly. Worth giving it a test! If the error comes back through nicely then less code is better code :D